pkg/pricing/aws/convert.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aws
+
+import (
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/vmware/purser/pkg/controller/dgraph"
+	"github.com/vmware/purser/pkg/controller/dgraph/models"
+)
+
+const (
+	na              = "na"
+	aws             = "aws"
+	gbMonth         = "GB-Mo"
+	deliminator     = "-"
+	storageInstance = "Storage"
+	computeInstance = "Compute Instance"
+)
+
+// GetRateCardForAWS takes region as input and returns RateCard and error if any
+func GetRateCardForAWS(region string) (*models.RateCard, error) {
+	awsPricing, err := GetAWSPricing(region)
+	if err != nil {
+		return nil, err
+	}
+	rateCard := convertAWSPricingToPurserRateCard(region, awsPricing)
+	return rateCard, nil
+}
+
+func convertAWSPricingToPurserRateCard(region string, awsPricing *Pricing) *models.RateCard {
+	nodePrice, storagePrice := getResourcePricesFromAWSPricing(awsPricing)
+	return &models.RateCard{
+		IsRateCard:    true,
+		CloudProvider: aws,
+		Region:        region,
+		NodePrices:    nodePrice,
+		StoragePrices: storagePrice,
+	}
+}
+
+func getResourcePricesFromAWSPricing(awsPricing *Pricing) ([]*models.NodePrice, []*models.StoragePrice) {
+	var nodePrices []*models.NodePrice
+	var storagePrices []*models.StoragePrice
+	products := awsPricing.Products
+	planList := awsPricing.Terms
+
+	duplicateComputeInstanceChecker := make(map[string]bool)
+	for _, product := range products {
+		if product.ProductFamily == computeInstance {
+			key := product.Sku + product.Attributes.InstanceType + product.Attributes.OperatingSystem
```

Comment:
Probably create a separate method for this for better readability.

Reply:
Done